test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,241 @@
+
+    final options = RequestsSignatureOptions(
+      clientId: 'test_client_id',
+      clientSecret: 'test_client_secret',
+      headerName: 'X-Signature',
+      signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}',
+      disableAutoRetryOnClockSkew: false,
+      clockSkew: Duration(milliseconds: clockskewMS),
+    );
+
+    final dio = Dio();
+    final dioAdapter = DioAdapter(dio: dio);
+    dio.httpClientAdapter = dioAdapter;
+
+    final serverDate = DateTime.now().toUtc().add(Duration(milliseconds: 6000));
+    final serverDateIsoString = serverDate.toIso8601String();
+
+    // Create the interceptor
+    final interceptor = RequestsSignatureInterceptor(
+      options,
+      dio,
+      getTime: (request) {
+        return DateTime.now().toUtc().millisecondsSinceEpoch ~/ 1000;
+      },
+      getDateHeader: () {
+        return serverDateIsoString;
+      },
+    );
+
+    // Add the interceptor to Dio
+    dio.interceptors.add(interceptor);
+
+    int requestCount = 0;
+
+    dioAdapter.onGet('https://google.ca/', (request) {
+      requestCount++;
+      request.reply(HttpStatus.ok, {});
+    });
+
+    // Act
+    final response = await dio.get('https://google.ca/');
+
+    // Assert
+    expect(response.statusCode, HttpStatus.ok);
+
+    // Calculate the expected time difference
+    final now = DateTime.now().toUtc();
+    final expectedDiff = serverDate.difference(now).inMilliseconds;
+
+    // Assert that the time difference equals the clock skew
+    expect(expectedDiff.abs(), lessThanOrEqualTo(clockskewMS + toleranceMS));
+
+    expect(requestCount, 1);
+  });
+
+  test('Auto-retry disnabled when no date header', () async {
+    final clockskewMS = 6000; // Clock skew in milliseconds
+    final toleranceMS = 1200; // Tolerance in milliseconds
+
+    final options = RequestsSignatureOptions(
+      clientId: 'test_client_id',
+      clientSecret: 'test_client_secret',
+      headerName: 'X-Signature',
+      signaturePattern: '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}',
+      disableAutoRetryOnClockSkew: false,
+      clockSkew: Duration(milliseconds: 6000),
+    );
+
+    final dio = Dio();
+    final dioAdapter = DioAdapter(dio: dio);
+    dio.httpClientAdapter = dioAdapter;
+
+    final serverDate = DateTime.now().toUtc().add(Duration(milliseconds: 6000));
+
+    // Create the interceptor
+    final interceptor = RequestsSignatureInterceptor(options, dio);
+
+    // Add the interceptor to Dio
+    dio.interceptors.add(interceptor);
+
+    int requestCount = 0;
+
+    dioAdapter.onGet('https://google.ca/', (request) {
+      requestCount++;
+      request.reply(HttpStatus.ok, {});
+    });
+
+    await dio.get('https://google.ca/');
+
+    // Calculate the expected time difference
+    final now = DateTime.now().toUtc();
+    final expectedDiff = serverDate.difference(now).inMilliseconds;
+
+    // Assert that the time difference equals the clock skew
```

Comment:
I don't understand this assertion, can you explain this to me? @thorvalld @jeanplevesque

Reply:
to check if the difference between actual time and the expected time of response after taking in the clock skew checks out

Reply:
But it looks like you are testing that the auto retry is disabled in this test because their is no date time header so why  are we playing around with times, its confusing me a little bit :')

Reply:
correct, I dropped the unnecessary use of that assertion